src/functions_math.jl

```diff
@@ -127,3 +132,152 @@ function symmetric_names(L::Tuple{Symbol,Symbol}, dims::Integer)
     end
     return compile_time_return_hack(names)
 end
+
+################################################
+# FFT
```

Comment:
Maybe we should move this all to its own file?
and then the test to match?

Reply:
Done.